spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageProducingHandler.java

```diff
@@ -150,8 +167,30 @@ else if (reply instanceof AbstractIntegrationMessageBuilder) {
 			}
 		}
 
-		Message<?> replyMessage = createOutputMessage(reply, requestHeaders);
-		sendOutput(replyMessage, replyChannel);
+		if (this.asyncReplySupported && reply instanceof ListenableFuture<?>) {
+			ListenableFuture<?> future = (ListenableFuture<?>) reply;
+			final Object fallbackReplyChannel = replyChannel;
```

Comment:
Well, it really is a "fall back" channel - i.e. derived from the headers somehow - we never send to this channel if an explicit `outputChannel` exists.

But I am ok with `theReplyChannel`.